src/gluonts/dataset/tabular.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Iterator, List, Union, Dict
+
+import pandas as pd
+
+from gluonts.dataset.common import Dataset, DataEntry, ProcessDataEntry
+from gluonts.dataset.field_names import FieldName
+
+
+@dataclass
+class DataFramesDataset(Dataset):
+    """
+    A pd.DataFrame-based dataset type.
+
+    This class is constructed with a collection of pd.DataFrame-objects where each
+    DataFrame is representing one time series. A target and a timestamp columns are
+    essential. Furthermore, static/dynamic real/categorical features can be specified.
+
+    Parameters
+    ----------
+    dataframes: Union[List[pd.DataFrame], Dict[str, pd.DataFrame]]
+        List or Dict of 'pd.DataFrame's containing at least timestamp and target columns.
+        If a Dict is provided, the key will be the associated 'item_id'.
```

Comment:
Double-backticks to get `monospace` font in sphinx-generated docs

```suggestion
        List or Dict of ``pd.DataFrame``s containing at least timestamp and target columns.
        If a Dict is provided, the key will be the associated ``item_id``.
```